app/models/concerns/findings/finding_answers_generator.rb

```diff
@@ -0,0 +1,78 @@
+# require 'app/extension/extended_email_reply_parsers/spanish_parser'
+
+module Findings::FindingAnswersGenerator
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def receive_finding_answers
+      config
+
+      Mail.all.each do |mail|
+        begin
+          receive_mail mail
+        rescue Exception => exception
+          log exception, mail
+        end
+      end
+    end
+
+    private
+
+    def receive_mail mail
+      finding_id = extract_finding_id mail
+
+      if finding_id && exists?(finding_id)
+        find(finding_id).finding_answers.create generate_finding_answer_from_mail(mail)
+      else
+        NotifierMailer.notify_not_added_answer(mail.from, extract_answer(mail)).deliver_later
+      end
+    end
+
+    def config
+      Mail.defaults do
+        retriever_method ENV['EMAIL_METHOD'].to_sym,  address:    ENV['EMAIL_SERVER'],
```

Comment:
El método lo pondría en un método aparte (valga la redundancia), para que quede algo como:

```suggestion
        retriever_method email_method, address:    ENV['EMAIL_SERVER'],
```
Sino se ve bastante raro.

Reply:
okey, lo dejo en un metodo a parte que solo consule la variable de entorno? ahi te lo subo y me decis

Reply:
Exacto, esto mismo, el método debería ser privado y algo como:

```ruby
  def email_method
    ENV['EMAIL_METHOD'].to_sym
  end
```

Y viendo otra cosa, deberíamos agregar uno que sea:

```ruby
  def email_method?
    ENV['EMAIL_METHOD']
  end
```

Así preguntamos antes de hacer todo el resto si hay una config, porque nos va a pasar (al menos al principio), que la gran mayoría de los clientes no lo va a tener puesto.

Reply:
perfecto. Y en el caso de que email_method? de falso, no hago nada?

Reply:
Exacto, pasamos de largo si no está configurado.

Reply:
perfecto franco. Te consulto, vos sabes que si saco afuera el metodo email_method en el bloque de retriever_method cuando lo llamo me dice que no esta definido para Mail::Configuration. Debe ser porque esta en ese bloque. Decis que lo deje como publico en este modulo y lo llame como Finding.email_method? o dejo directo la variable de entorno?

Reply:
Ahh claro, eso está en un bloque medio raro de config... fijate de ponerlo a la misma altura que config, ahí en teoría debería funcionar.

Reply:
perfecto franco. Ahi rubocop me sugirio una forma definiendo un lambda decime que te parece y si queres lo dejo asi.